core/src/log/split.rs

```diff
@@ -0,0 +1,74 @@
+use std::fmt::Debug;
+use std::result::Result as StdResult;
+
+use graph::slog::*;
+use graph::slog_async;
+
+/// An error that could come from either of two slog `Drain`s.
+#[derive(Debug)]
+enum SplitDrainError<D1, D2>
+where
+    D1: Drain,
+    D2: Drain,
+{
+    Drain1Error(D1::Err),
+    Drain2Error(D2::Err),
+}
+
+/// An slog `Drain` that forwards log messages to two other drains.
+struct SplitDrain<D1, D2>
+where
+    D1: Drain,
+    D2: Drain,
+{
+    drain1: Box<D1>,
+    drain2: Box<D2>,
```

Comment:
This `SplitDrain` is really elegant, but why the `Box`es?

Reply:
Don't I need those because `D1` and `D2` have unknown sizes at compile time?

Reply:
The `Sized` bound, which guarantees the type (whatever type it is) has a known size at compile time, is implicit.